### ISSUE TITLE
Material Grid layout issues

### ISSUE THREAD

**Issue by remarkNS** [GwtMaterialDesign/gwt-material#62]
Material grid is now much better than the previous version but there are still some issues:

1. Grid does not render the same on IE (ver. 11) and Chrome (ver. 44).

a.) On IE (on the left) there are (bold) lines that separates header items (in this case rows) that is nice I think - no such lines on Chrome. On Chrome there are line that separates header with table content that is also nice I think - no such lines on IE. I am not sure but maybe row with checkboxes should also be separated with the line from the rest of table.

b.) On IE "Item name" does not wrap on next line and it is much better than Chrome version that wraps because all rows bellow it are not aligned correctly (look Price, Action).

c.) On IE there is a Vertical scroll bar that is not needed, on Chrome it is OK

![ie_chrome](https://cloud.githubusercontent.com/assets/9980558/8869919/bf71a9fa-31e9-11e5-880f-8faa3283e222.png)

2. Bordered style not rendered correctly.

If Stripped style is rendered like this:

![striped](https://cloud.githubusercontent.com/assets/9980558/8869930/e46f7cc8-31e9-11e5-984e-a7603dbaf30f.png)

than I think Bordered should look like this (not with red color off course):

![bordered](https://cloud.githubusercontent.com/assets/9980558/8869944/11ba2f2a-31ea-11e5-81a2-d3dff3f52518.png)

3. Still some layout issues on mobile:

Unnecessary scroll bars of outer panel ? shown:

![m1](https://cloud.githubusercontent.com/assets/9980558/8869964/4f61c126-31ea-11e5-8c8e-fad719faae20.png)

Unnecessary scroll bars of outer panel ? shown:

![m2](https://cloud.githubusercontent.com/assets/9980558/8869975/6ac5e280-31ea-11e5-87e2-691c71eac416.png)

Last row has been cut ? This is a case when outer panel did not register that it should scroll ?

![m3](https://cloud.githubusercontent.com/assets/9980558/8870004/b33d6ed4-31ea-11e5-9115-1261d5361555.png)

Note: I think rendering on Mozilla FF is like on Chrome.

**Comment by slugmandrew** [comment]
Thanks for this very comprehensive bug report. We're looking into it and appreciate the feedback.

**Comment by slugmandrew** [comment]
@fmax32 Is this your area of expertise?

**Comment by fmax32** [comment]
@slugmandrew - almost), but I will do my best to fix it! Thanks!

**Comment by remarkNS** [comment]
One small addition: show different background color for selected row like described in https://www.google.com/design/spec/components/data-tables.html#

![selected](https://cloud.githubusercontent.com/assets/9980558/9003156/a867c582-376b-11e5-8070-e40f982200b6.png)

**Comment by BenDol** [comment]
The GWT DataGrid isn't responsive and it's designed poorly for responsiveness as far as I can tell. Might need to create our own datagrid based on another implementation.

**Comment by remarkNS** [comment]
I think Ben is right, another problem with Gwt DataGrid is that it has no grouping, it does not support hierarchy (TreeGrid), does not have any kind of aggregations (SUM or whatever) or even basic thing like hiding, resizing or reordering columns. Without fast and capability rich DataGrid component this (or anything else, not only this library) can hardly be used for any serious business purpose. I can not imagine any Banking, ERP or even CRM type of  software without this type of Grid component. Vaadin Grid looks promising but it still missing many of previously stated. Maybe we can open one Issue and discuss about this ? Maybe someone has an idea or knows some great open source JS component that we can wrap and incorporate in this great looking library. Maybe Vaadin Grid is the best option in this moment, I don't know. What do you think guys ? Does anybody else think that this is important or not ? The type of Grid I am talking about is Sencha Grid (that you can find here https://www.sencha.com/examples/) but it is not really open source and it has very unattractive commercial licence as well.

**Comment by dlaunen** [comment]
FYI, we're looking to solve this issue with a highly functional grid/data
table that we will contribute to gwt-material.  We already have compiled a
list of requirements, it might make sense to share those with the community
and if certain functionality is missing that the community might want they
can add feature requests.

David A. Launen
Vice President, Information Technology
Insurance Insight Inc.
Registered Insurance Brokerage/
Exclusive WALKAWAY Distributor
o. 416.603.8555 x130
c. 416.770.7495

**Comment by remarkNS** [comment]
@dlaunen Great news ! I am looking forward to see this. Please open one issue where we can discuss about this. I think that "highly functional grid/data table" as you said would be a huge step forward for gwt-material. If it's possible to have all or at least the most of functionalities stated in my previous post, with beautiful and responsive material design, that would be a "killer" feature.

**Comment by gilberto-torrezan** [comment]
Here is my approach to (kind-of) responsive DataGrids:

Consider this table:

![table-large](https://cloud.githubusercontent.com/assets/7464735/11251873/6d55fb70-8e1a-11e5-9653-463f1d2a1117.png)
 
Using CSS sorcery with converting it to this small table on mobile, without horizontal scroll:

![table-small](https://cloud.githubusercontent.com/assets/7464735/11251898/947253fc-8e1a-11e5-99a2-cb658da10f85.png)

The main trick here is to convert table cells into blocks:

```css
.dataGridEvenRowCell, .dataGridOddRowCell {
	border: none;
	display: block;
	height: auto !important;
}
```
That's what I'm using on my projects while a better Widget doesn't come out.

**Comment by kevzlou7979** [comment]
@gilberto-torrezan Thanks for the workaround, note that this will be just added as a fix on gwt-material 1.4.1 snapshot but on 1.5 there will be an ongoing development on our own DataGrid.

**Comment by kevzlou7979** [comment]
This is the reference for the discussion https://github.com/GwtMaterialDesign/gwt-material/issues/205
